### PR TITLE
Add unit tests for BeanMediatorFactory 

### PR DESCRIPTION
**Pull request by Nirothipan**
## Automation tests
 - Unit tests 
> 78%

### CODE REVIEW

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+     */
+    @Test
+    public void testCreateSpecificMediator() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"CREATE\" class=\"org.apache.synapse.util.TestTask\" " +
+                "var=\"loc\"></bean>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
```
Comment (Asitha) [inline modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Shall we correct this line segment as follows, to make it more readable?
```
Assert.assertTrue("BeanMediator is not created successfully.", 
                  mediator instanceof BeanMediator);
```

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    @Test
+    public void testCreateSpecificMediator2() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"SET_PROPERTY\" property=\"latitude\" value=\"{//latitude}\" var=\"loc\" " +
+                "xmlns:ns3=\"http://org.apache.synapse/xsd\"             " +
+                "xmlns:ns=\"http://org.apache.synapse/xsd\"></bean>\n";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
```
Comment (Asitha) [inline modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Shall we correct this line segment as follows, to make it more readable?

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    @Test
+    public void testCreateSpecificMediator3() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"GET_PROPERTY\" property=\"latitude\" value=\"{//latitude}\" var=\"loc\" " +
+                "target=\"target\"   xmlns:ns3=\"http://org.apache.synapse/xsd\"             " +
+                "xmlns:ns=\"http://org.apache.synapse/xsd\"></bean>\n";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
```
Comment (Asitha) [inline modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Shall we correct this line segment as follows, to make it more readable?

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    @Test
+    public void testCreateSpecificMediator4() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"REMOVE\" property=\"latitude\" value=\"{//latitude}\" var=\"loc\" " +
+                "xmlns:ns3=\"http://org.apache.synapse/xsd\"             " +
+                "xmlns:ns=\"http://org.apache.synapse/xsd\"></bean>\n";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
```
Comment (Asitha) [inline modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Shall we correct this line segment as follows, to make it more readable?

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java (hunk 1 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Fixed with https://github.com/wso2/wso2-synapse/pull/1053

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java (hunk 2 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Fixed with https://github.com/wso2/wso2-synapse/pull/1053

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java (hunk 3 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Fixed with https://github.com/wso2/wso2-synapse/pull/1053

File: modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java (hunk 4 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java]: Fixed with https://github.com/wso2/wso2-synapse/pull/1053

**Pull request merged**